lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -294,8 +308,29 @@ def write(chunk)
             end
           end
         end
+        if @running_cloudfunctions
+          # If the current group of entries is coming from a Cloud Functions
+          # function, the function name can be extracted from the tag.
+          match_data = @cloudfunctions_tag_regexp.match(tag)
+          if match_data
+            # Service name is set to Cloud Functions only for logs actually
+            # coming from a function.
+            @service_name = CLOUDFUNCTIONS_SERVICE
+            labels = write_log_entries_request['commonLabels']
+            labels["#{CLOUDFUNCTIONS_SERVICE}/function_name"] =
+              match_data['function_name']
+          else
+            # Other logs are considered as coming from the Container Engine
+            # service.
+            @service_name = CONTAINER_SERVICE
+          end
+        end
         arr.each do |time, record|
           next unless record.is_a?(Hash)
+          if @service_name == CLOUDFUNCTIONS_SERVICE && record.key?('log')
```

Comment:
is the format of 'log' set in stone or could you emit it as json instead?  I think we could eliminate some of the special-case code if so?